server/db/model/User.js

```diff
@@ -0,0 +1,151 @@
+var bcrypt = require('bcrypt'),
+    jwt = require('jsonwebtoken'),
+    mongoose = require('../index.js'),
+    config = require('../../../config/default.js'),
+    Schema = mongoose.Schema,
+    secret = config.secret;
+
+// Define the document Schema
+var userSchema = new Schema({
+    full_name: String,
+    email: {
+        type: String,
+        required: true,
+        index: {
+            unique: true
+        }
+    },
+    email_verified: Boolean,
+    password: {
+        type: String,
+        required: true
+    },
+    tokens: [{
+        token: String
+    }],
+    old_tokens: [{
+        token: String
+    }],
+    created_date: {
+        type: Date,
+        default: Date.now
+    },
+    birthday: Date,
+    groups: [{
+        name: String
+    }],
+    meta: {
+        ip: String
+    }
+});
+
+// Allow us to query by name
+userSchema.query.byName = function(name) {
+    return this.findOne({
+        full_name: new RegExp(name, 'i')
+    });
+};
+
+// Allow us to query by email
+userSchema.query.byEmail = function(email) {
+    return this.findOne({
+        email: new RegExp(email, 'i')
+    });
+};
+
+// Allow us to query by token
+userSchema.query.byToken = function(findToken) {
+    return this.findOne({
+        tokens: {
+            $elemMatch: {
+                token: findToken
+            }
+        }
+    });
+};
+
+// Verify the token is correct
+userSchema.methods.verifyToken = function(token) {
+    try {
+        tokenData = jwt.verify(token, secret);
+        if (this.email == tokenData.email) {
+            return [true, null];
+        } else {
+            return [false, 'Email does not match token'];
+        }
+    } catch (err) {
+        console.error(err);
+        switch(err.name) {
+            case 'TokenExpiredError':
```

Comment:
When a token is expired do we have to handle anything here to get a new one or is that somewhere else